### PR TITLE
fixes body array params

### PR DESCRIPTION
**Pull request by LeFnord**
… corrects the setting of the type for parameters which are given as array, adds missing format

### CODE REVIEW

File: lib/grape-swagger/doc_methods/move_params.rb [hunk 1]
```diff
@@ -113,6 +118,17 @@ def build_definition(name, verb = nil)
           name
         end
 
+        def prepare_nested_types(params)
+          params.each do |param|
+            param.tap do |x|
```
Comment (dblock) [inline lib/grape-swagger/doc_methods/move_params.rb]: This seems silly, all this does is rename `param` into `x`, wjy don't you just do `do |x|` above or, better leave it as `param`?

File: lib/grape-swagger/doc_methods/move_params.rb [hunk 2]
```diff
@@ -145,10 +161,7 @@ def property_keys
         def movable?(param)
           param[:in] == 'body'
         end
-
-        def deletable?(param)
-          param[:in] == 'body'
-        end
+        alias deletable? movable?
```
Comment (dblock) [inline lib/grape-swagger/doc_methods/move_params.rb]: Aren't you supposed to make those symbols?, like `:deletable?` and `:movable?`?

**Comment by dblock** [comment]
My `tap` comment makes sense I think, the rest is fine, merge at will.

File: lib/grape-swagger/doc_methods/move_params.rb (hunk 1 above)
Comment (LeFnord) [inline lib/grape-swagger/doc_methods/move_params.rb]: yeap, would be changed, also in `prepare_nested_names`

File: lib/grape-swagger/doc_methods/move_params.rb (hunk 2 above)
Comment (LeFnord) [inline lib/grape-swagger/doc_methods/move_params.rb]: this is a rubocop recommendation

Comment (dblock) [inline lib/grape-swagger/doc_methods/move_params.rb]: Oh that's 2.x syntax, mind blown.

**Comment by dblock** [comment]
Should we be worried that things like `nested_definitions` are modifying `params`?

**Comment by LeFnord** [comment]
only if they did … but here only the params given in body, would be moved/translated into a definition object, this could be seen as a direct JSON specification of the request parameters for POST/PUT/PATCH requests

**Pull request merged**